File: ChatApi_OpenRouter/ChatApi/Models/ChatModels.cs
namespace ChatApi.Models;

public record ChatMessage(string Role, string Content);

/// <summary>
/// Request body for both non-streaming and streaming chat endpoints.
/// The optional Model field lets the client override the default model
/// configured in appsettings.json.
/// </summary>
public record ChatRequest(
    IReadOnlyList<ChatMessage> Messages,
    string? Model = null);

public record ChatResponse(string Reply);

/// <summary>
/// Represents a single model available on OpenRouter.
/// Returned by GET /api/chat/models.
/// </summary>
public record ModelInfo(string Id, string Name, string? Description = null);

File: chat-ui_OpenRouter/src/app/models/chat.models.ts
export interface ChatMessage {
  role: 'user' | 'assistant' | 'system';
  content: string;
}

export interface ChatRequest {
  messages: ChatMessage[];
  model?: string;
}

export interface ChatResponse {
  reply: string;
}

// ─── Chat Session (localStorage history) ───────────────────────

export interface ChatSession {
  id: string;
  title: string;
  model: string;
  messages: ChatMessage[];
  createdAt: string;      // ISO date string
  updatedAt: string;      // ISO date string
}

// ─── Available Model (from GET /api/chat/models) ───────────────

export interface AvailableModel {
  id: string;
  name: string;
  description?: string;
}

// ─── Voice Input State ─────────────────────────────────────────

export enum VoiceState {
  Idle        = 'idle',
  Listening   = 'listening',
  Processing  = 'processing',
  Unsupported = 'unsupported',
}
